src/rest.js

```diff
@@ -35,7 +35,8 @@ export class Rest {
   request(method, path, body, options = {}) {
     let requestOptions = extend(true, {}, this.defaults, options);
 
-    requestOptions.method = method;
+    // Force method to be upper-case for pre-flight Accept-Methods matching
+    requestOptions.method = method.toUpperCase();
```

Comment:
I didn't :p I think this should be solved server-side. The thing is, this small stupid change is a BC-break. With the new major version coming up, I'd say we can go with this if it brings happiness :D